examples/plot_intersectional_bias.py

```diff
@@ -0,0 +1,1327 @@
+# of which may be disproportionately affecting one subgroup of the population defined by at least
+# two intersecting features that we know of (gender and race).
+#
+# The hypothetical scenario
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Based on this finding, we’ve developed a hypothetical scenario. You should note that this
+# scenario (and the simulated data we’re using) is **overly simplified**. But we hope it gives you
+# a sense of how the application of ML can exacerbate biases in training data, and how we can
+# evaluate models for bias from an intersectional perspective.
+#
+# Imagine we have some **electronic health record data on 10,000 patients** who have been diagnosed
+# at *Health System A* with either affective disorder (denoted with 0) or schizophrenia (denoted
+# with 1) over the past 10 years. This data contains information on their sex, race, some
+# psychosocial information, and information from clinical assessments on their symptoms. Hospital
+# executives have had a classifier built that will take this information collected about a patient
+# at intake, and **assign a diagnosis of either affective disorder (0) or schizophrenia (1)**. They
+# train a binary classifier on the data to assign new incoming patients a diagnosis, to triage them
+# into the appropriate clinic for treatment.
+#
+# Now, let’s say this hospital is run by some executives who would really like to cut costs. They
+# don’t want to do any further assessment after classification, and they are **planning to
+# administer treatments to patients following this automated triage.**
+#
+# But first, they must provide some evidence to stakeholders that this classifier works well in
+# diagnosing patients, so they have a test sample of 1000 patients, on which they must evaluate the
+# model. They argue that their classifier works very well, based on some impressive sensitivity and
+# specificity values. The stakeholders are not fully convinced (thinking that the executives may be
+# a little too eager to get this model deployed), so they hire us (an independent consulting firm)
+# to evaluate the model.
+#
+# Note that this is one overly simplistic component of our case scenario. Typically, much more
+# consideration and evaluation would occur before ML is deployed in a real-life setting, especially
+# in healthcare. When it comes to complicated diagnoses (like schizophrenia), any use of ML is also
+# likely to complement human or clinical judgment.
+#
+#
+# Model Development
+# =================
+#
+# Let's build our first machine learning model. First, we need to import the required libraries and
+# data set.
+#
+
+# %%
+
+# Import relevant libraries
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import sklearn.metrics as skm
+import seaborn as sns
+from sklearn.datasets import fetch_openml
+
+# Read in dataset
+data_openml = fetch_openml(data_id=45040)
```

Comment:
Is this a custom dataset? Suggest adding a sentence to describe its provenance (but this might be unnecessary)

Reply:
It's a synthetic dataset generated by Marta using a semi-manual approach, so perhaps adding a description may not lend much additional clarity.

Reply:
Fair, I am resolving this